### PR TITLE
[6.1] feat/style: check malformed character in java source

### PR DESCRIPTION
**Pull request by miurahr**
Allowing UTF-8 in Java code has a risk of malformded commits that is hard to detect in review process.
It is known as “Trojan Source” Attack.
[The answer is proper review](https://research.swtch.com/trojan#review)
Making LTR/RTL code points visible is particularly important for review tools. The proposal here
is improve review process to check dangerous character in source code when Pull-Request.

The PR also update developer note about coding sytle.

## Pull request type

- Build and release changes -> [build/release]

## What does this PR change?

- test: check a existence of Right-to-Left-Override "\u202e" character that is used to inject malformded code in the code
- test: check a existence of invisible characters such as  TAB, Zenkaku-Space, Non breakable space in the coee

## Other information

- Please see https://attack.mitre.org/techniques/T1036/002/
for detailed attack procedure description.

- Please see GitHub warns RtL text
https://github.blog/changelog/2021-10-31-warning-about-bidirectional-unicode-text/

- Thesis: "Trojan Source: Invisible Vulnerabilities" https://trojansource.codes/trojan-source.pdf

### CODE REVIEW

File: test/src/svn/BundleTest.java [hunk 1]
```diff
@@ -196,17 +212,33 @@ public static void processSourceContent(BiConsumer<Path, CharSequence> consumer)
...
+    public static void checkFileContent(Path p, CharsetDecoder decoder,
+                                        BiConsumer<Path, CharSequence> consumer) throws Exception {
+        try {
+            byte[] bytes = Files.readAllBytes(p);
+            CharBuffer chars = decoder.decode(ByteBuffer.wrap(bytes));
+            for (int i =0; i < chars.limit(); i++) {
+                int c = chars.charAt(i);
+                if (c == 0x202e) {
+                    // found Right-to-left-override: unicode 202e
+                    throw new Exception("File contains Right-to-Left-Override (RLTO) character: " + p);
```
Comment (miurahr) [inline test/src/svn/BundleTest.java:233]:
```suggestion
                if (0x2028 <= c && c < 0x202f) {
                    // found Right-to-left control characters: most dangerous RtLO is 0x202e
                    throw new Exception("File contains Right-to-Left control character: " + p);
```

File: docs_devel/docs/30.CodingStyles.md [hunk 2]
```diff
@@ -36,11 +46,11 @@ You can check your code with;
 You can check your code with Checkstyle linting tool.
 A project configuration is stored in `config/checkstyle/checkstyle.xml`
 We treat checkstyle lint error as warning, but you are recommended to
-improve your code before sending your patch to core team.
+improve your code before sending your patch to a core team.
```
Comment (brandelune) [inline docs_devel/docs/30.CodingStyles.md:49]: to *the* core team ?

**Review by brandelune — CHANGES_REQUESTED** [review]

File: docs_devel/docs/30.CodingStyles.md [hunk 3]
```diff
@@ -2,26 +2,36 @@
...
+  * If you want, you are recommended to implement it as a plugin.
 * We move XML parser to be StAX based instead of JAXB because of performance.
   * We use [Jackson](https://github.com/FasterXML/jackson) XML serialization library.
 
+## language and characters
+
+- OmegaT project uses UTF-8 characters for comments in java source file in general.
+- You should use English for comment language.
+- You should use ASCII characters and 7bit clean for variable names, method names et. al.
+- You are allowed to use String laterals in UTF-8 encoding.
```
Comment (t-cordonnier) [inline docs_devel/docs/30.CodingStyles.md:29]: Do you mean litterals?

File: docs_devel/docs/30.CodingStyles.md (hunk 2 above)
Comment (t-cordonnier) [reply docs_devel/docs/30.CodingStyles.md:49]: I agree, there is actually only one core team unless we later decide to delegate review depending on features or modules

File: src/org/omegat/gui/editor/SegmentBuilder.java [hunk 4]
```diff
@@ -291,13 +301,15 @@ private void createActiveSegmentElement(TMXEntry trans) throws BadLocationExcept
             posSourceLength = sourceText.length();
 
             if (trans.isTranslated()) {
-                //translation exist
+                // translation exist
```
Comment (t-cordonnier) [inline src/org/omegat/gui/editor/SegmentBuilder.java:304]: exists

**Review by t-cordonnier — APPROVED** [review]

File: docs_devel/docs/30.CodingStyles.md (hunk 3 above)
Comment (miurahr) [reply docs_devel/docs/30.CodingStyles.md:29]: yes, it is typo.

**Review by briacp — APPROVED** [review]
w/ Thomas suggestions

**Review by brandelune — APPROVED** [review]

**Pull request merged**